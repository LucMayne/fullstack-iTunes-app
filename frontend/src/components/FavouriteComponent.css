.saved-item-container > * {
    border: 2px solid;
    width: 300px;
    height: 360px;
    margin: 10px;
}

.save-image-type-container {
    display: flex;
    align-items: center;
}

.save-image-type-container img {
    padding-right: 0.7vw;
    height: 100px;
    margin-right: 10px;
}

.saved-item {
    font-family: 'Nova Flat';
    position: relative;
    padding: 1.5rem;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    background-color: rgba(126, 184, 235, 0.589);
}

/* disable the scrolling bar for the <p> text */
.save-name-item {
    font-family: 'Raleway';
    font-weight: 700;
    padding-bottom: 5vh;
    margin-top: 2vh;
    overflow-y: scroll;
    /* For Firefox */
    scrollbar-width: none; 
    /* For Internet Explorer and Edge */
    -ms-overflow-style: none;
}

/* hide the scrollbar */
.save-name-item::-webkit-scrollbar {
    width: 0; 
}

.description-item {
    font-family: 'Raleway';
    font-weight: 700;
    /* margin-top: 0.5vh; */
    margin-bottom: 2.8vh;
    overflow-y: scroll;
    /* For Firefox */
    scrollbar-width: none; 
    /* For Internet Explorer and Edge */
    -ms-overflow-style: none;
}

.description-item::-webkit-scrollbar {
    width: 0; 
}

.save-bottom-items {
    display: flex;
    padding-bottom: 1vh;
    position: absolute;
    bottom: 0;
}

.save-bottom-items a {
    margin-right: 8.5vw;
}

.save-bottom-items button {
    font-size: 15px;
    border: 0;
    border-radius: 8px;
    min-width: 4vw;
}

