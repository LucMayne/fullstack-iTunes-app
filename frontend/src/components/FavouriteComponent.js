import React from "react";
import './FavouriteComponent.css';

class FavouriteComponent extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      // display each favourite item with an image, type, name, url, and remove button
      <div className='saved-item-container'>
        {this.props.favourites.map((item, index) => (
          <div className='saved-item' key={index}>
              <div className='save-image-type-container'>
                <img src={item.coverArt} alt={index}></img>
                <p>{
                    item.kind === 'feature-movie' ? "movie" : item.kind
                }</p>
              </div>
              <p className='save-name-item'>{item.name}</p>
              {/* .description contains HTML tags... use dangerouslySetInnerHTML to change the html directly */}
              {(item.description ? <p className='description-item'>Description: <span dangerouslySetInnerHTML={{__html: item.description}}></span></p> : "")}

              <div className="save-bottom-items">
                <a href={item.viewURL} target='_blank' rel="noopener noreferrer">Link</a>
                {/* call a function that is passed in as a prop, to remove a favourite item */}
                <button onClick={() => this.props.onRemoveFavourite(item)}>Remove</button>
              </div>
          </div>
        ))}
      </div>
    );
  }
}

export default FavouriteComponent;

// websites used
// used this site for the dangerouslySetInnerHTML: https://blog.logrocket.com/using-dangerouslysetinnerhtml-in-a-react-application/
  