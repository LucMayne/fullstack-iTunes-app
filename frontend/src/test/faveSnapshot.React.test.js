import React from 'react';
import renderer from 'react-test-renderer';
import FavouriteComponent from '../components/FavouriteComponent';

test('FavouriteComponent renders correctly', () => {

    // array that contains favourites
    const favourites = [
        {
            name: "Harry Potter and the Sorcerer's Stone",
            kind: "feature-movie",
            viewURL: "https://itunes.apple.com/us/movie/harry-potter-and-the-sorcerers-stone/id271469503?uo=4",
            coverArt: "https://is1-ssl.mzstatic.com/image/thumb/Video126/v4/68/0a/36/680a3626-b96c-4d04-0bd6-859ecf79299b/pr_source.jpg/100x100bb.jpg",
            description: "In this enchanting film adaptation of J.K. Rowling's delightful bestseller, Harry Potter learns on his 11th birthday that he is the orphaned first son of two powerful wizards and possesses magical powers of his own. At Hogwarts School of Witchcraft and Wizardry, Harry embarks on the adventure of a lifetime. He learns the high-flying sport Quidditch and plays a thrilling game with living chess pieces on his way to face a Dark Wizard bent on destroying him. For the most extraordinary adventure, see you on platform nine and three quarters!"
        }
    ];

    // check if the snapshots match
    const tree = renderer
        .create(<FavouriteComponent favourites={favourites} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
