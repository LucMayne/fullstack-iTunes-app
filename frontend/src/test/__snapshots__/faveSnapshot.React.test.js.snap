// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouriteComponent renders correctly 1`] = `
<div
  className="saved-item-container"
>
  <div
    className="saved-item"
  >
    <div
      className="save-image-type-container"
    >
      <img
        alt={0}
        src="https://is1-ssl.mzstatic.com/image/thumb/Video126/v4/68/0a/36/680a3626-b96c-4d04-0bd6-859ecf79299b/pr_source.jpg/100x100bb.jpg"
      />
      <p>
        movie
      </p>
    </div>
    <p
      className="save-name-item"
    >
      Harry Potter and the Sorcerer's Stone
    </p>
    <p
      className="description-item"
    >
      Description: 
      <span
        dangerouslySetInnerHTML={
          {
            "__html": "In this enchanting film adaptation of J.K. Rowling's delightful bestseller, Harry Potter learns on his 11th birthday that he is the orphaned first son of two powerful wizards and possesses magical powers of his own. At Hogwarts School of Witchcraft and Wizardry, Harry embarks on the adventure of a lifetime. He learns the high-flying sport Quidditch and plays a thrilling game with living chess pieces on his way to face a Dark Wizard bent on destroying him. For the most extraordinary adventure, see you on platform nine and three quarters!",
          }
        }
      />
    </p>
    <div
      className="save-bottom-items"
    >
      <a
        href="https://itunes.apple.com/us/movie/harry-potter-and-the-sorcerers-stone/id271469503?uo=4"
        rel="noopener noreferrer"
        target="_blank"
      >
        Link
      </a>
      <button
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
